lib/tasks/git-init.js

```diff
@@ -19,20 +20,15 @@ module.exports = class GitInitTask extends Task {
       return Promise.resolve();
     }
 
-    let gitVersionWorked = false;
-    return this._gitVersion().then(() => {
-      gitVersionWorked = true;
-
-      return this._gitInit()
-        .then(() => this._gitAdd())
-        .then(() => this._gitCommit())
-        .then(() => ui.writeLine(chalk.green('Successfully initialized git.')));
-    })
-      .catch(error => {
-        if (gitVersionWorked) {
-          throw error;
+    return this._gitVersion()
+      .then(() => true, () => false)
```

Comment:
I wanted to move `.then(() => true, () => false)` to `this._gitVersion ` and rename `_gitVersion` => `_hasGit`, but then had to update test :P